core/domain/topic_jobs_one_off.py

```diff
@@ -92,3 +95,56 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class PopulateTopicThumbnailSizeOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to populate the thumbnail_size_in_bytes attribute in topic
+     models.
+     """
+
+    _DELETED_KEY = 'topic_deleted'
+    _ERROR_KEY = 'update_error'
+    _SUCCESS_KEY = 'thumbnail_size_updated'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [topic_models.TopicModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            yield (PopulateTopicThumbnailSizeOneOffJob._DELETED_KEY, 1)
+            return
+
+        if item.thumbnail_size_in_bytes is None:
+
+            file_system_class = fs_services.get_entity_file_system_class()
+            fs = fs_domain.AbstractFileSystem(file_system_class(
+                feconf.ENTITY_TYPE_TOPIC, item.id))
+
+            filepath = '%s/%s' % (
+                constants.FILENAME_PREFIX_THUMBNAIL, item.thumbnail_filename)
+
+            if fs.isfile(filepath):
+                item.thumbnail_size_in_bytes = len(fs.get(filepath))
+                item.put()
+                yield (
+                    PopulateTopicThumbnailSizeOneOffJob._SUCCESS_KEY, 1)
+            else:
+                yield (
+                    PopulateTopicThumbnailSizeOneOffJob._ERROR_KEY,
+                    'Thumbnail %s for topic %s not found on the filesystem' % (
+                        item.thumbnail_filename,
+                        item.id
+                    ))
+        else:
+            # The attribute thumbnail_size_in_bytes is already updated.
+            yield (
+                PopulateTopicThumbnailSizeOneOffJob._SUCCESS_KEY, 1)
```

Comment:
Have different keys for "already populated" and "newly populated" -- i.e. distinguish the two "success" cases in some way. (This would then, e.g., allow re-running until "newly populated" is zero.)

Reply:
Done.